tools/common_lib/src/layers_utils.h

```diff
@@ -351,6 +351,67 @@ inline void fill_with_constant_linear_container_half(std::span<std::byte> contai
     }
 }
 
+inline uint8_t clamp_value(uint8_t value, uint8_t min, uint8_t max)
+{
+    if (value < min) return min;
+    else if (value > max) return max;
+    else return value;
+}
+
+// This function is not taking care of unaligned sizes
+inline void fill_quantized_data_half_to_uint4(std::span<std::byte> output_data, std::span<std::byte> input_data, uint32_t chunk_size, std::span<std::byte> output_scale, std::span<std::byte> output_zeropoint)
+{
+    using Dt = Half;
+    auto limit = (2 << 3) - 1;
+
+    auto* input_ptr = reinterpret_cast<Dt*>(input_data.data());
```

Comment:
const auto, we dont want to modify input data by accident

Reply:
Done